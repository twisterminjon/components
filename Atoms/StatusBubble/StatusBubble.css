.statusbubble {
  position: relative;
  height: 11px;
  width: 11px;
  border-radius: 50%;

  border: 1px solid var(--statusbubble__border);
}

.statusbubble--available {
  background-color: var(--statusbubble_available);
}

.statusbubble--busy {
  background-color: var(--statusbubble_busy);
}

.statusbubble--away {
  background-color: var(--statusbubble_away);
}

.statusbubble--offline {
  background-color: var(--statusbubble_offline);
}
