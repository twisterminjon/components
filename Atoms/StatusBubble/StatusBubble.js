import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './StatusBubble.css'
import { USER_STATUS_LIST } from '../../../constants'

export default class StatusBubble extends Component {
  static propTypes = {
    /** The status to to display in the bubble */
    status: PropTypes.oneOf(USER_STATUS_LIST).isRequired,
    /** ClassName for the wrapper */
    className: PropTypes.string,
    /** Style for the wrapper */
    style: PropTypes.object,
  }
  static defaultProps = {
    className: '',
    style: {},
  }

  render() {
    const { status, className, style } = this.props

    const colorClass = `statusbubble--${status}`

    return <div className={`statusbubble ${colorClass} ${className}`.trim()} style={style} />
  }
}
