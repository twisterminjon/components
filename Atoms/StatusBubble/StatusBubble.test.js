import React from 'react'
import StatusBubble from './StatusBubble'

describe('StatusBubble', () => {
  it('matches the snapshot - available', () => {
    const wrapper = window.shallow(<StatusBubble status="available" />)
    expect(wrapper).toMatchSnapshot()
  })

  it('matches the snapshot - busy', () => {
    const wrapper = window.shallow(<StatusBubble status="busy" />)
    expect(wrapper).toMatchSnapshot()
  })

  it('matches the snapshot - offline', () => {
    const wrapper = window.shallow(<StatusBubble status="offline" />)
    expect(wrapper).toMatchSnapshot()
  })
})
