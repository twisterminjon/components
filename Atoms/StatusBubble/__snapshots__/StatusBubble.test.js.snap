// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBubble matches the snapshot - available 1`] = `
<div
  className="statusbubble statusbubble--available"
  style={Object {}}
/>
`;

exports[`StatusBubble matches the snapshot - busy 1`] = `
<div
  className="statusbubble statusbubble--busy"
  style={Object {}}
/>
`;

exports[`StatusBubble matches the snapshot - offline 1`] = `
<div
  className="statusbubble statusbubble--offline"
  style={Object {}}
/>
`;
